src/app/clusters/bindings/bindings.cpp

```diff
@@ -20,155 +20,265 @@
+                    {
+                        ChipLogError(Zcl,
+                                     "Binding: Failed to remove pending notification for unicast binding " ChipLogFormatX64
+                                     ": %" CHIP_ERROR_FORMAT,
+                                     ChipLogValueX64(entry.nodeId), err.Format());
+                    }
+                }
+            }
+        }
+        emberClearBinding();
+
+        // Add entries currently not in the binding table
+        uint8_t bindingIndex = 0;
+        for (auto iter = newBindingList.begin(); iter.Next();)
+        {
+            uint8_t oldIndex = 0;
+            if (iter.GetValue().nodeId.HasValue() && IsInBindingTable(path.mEndpointId, iter.GetValue(), &oldIndex))
+            {
+                BindingManager::GetInstance().UnicastBindingMoved(oldIndex, bindingIndex);
+            }
+            else if (iter.GetValue().nodeId.HasValue())
+            {
+                err =
+                    BindingManager::GetInstance().UnicastBindingCreated(iter.GetValue().fabricIdx, iter.GetValue().nodeId.Value());
+                if (err != CHIP_NO_ERROR)
+                {
+                    ChipLogProgress(Zcl,
+                                    "Binding: Failed to create session for unicast binding to device " ChipLogFormatX64
+                                    ": %" CHIP_ERROR_FORMAT,
+                                    ChipLogValueX64(iter.GetValue().nodeId.Value()), err.Format());
+                }
+            }
+            bindingIndex++;
+        }
+        for (auto iter = newBindingList.begin(); iter.Next();)
+        {
+            AddBindingEntry(iter.GetValue(), path.mEndpointId);
+        }
+        return err;
+    }
+    else if (path.mListOp == ConcreteDataAttributePath::ListOperation::AppendItem)
     {
-        ChipLogError(Zcl, "Binding: Failed to remove pending notification for unicast binding" ChipLogFormatX64 ": %s",
-                     ChipLogValueX64(nodeId), err.AsString());
+        printf("append\n");
+        DecodableBindingListType newBindingList;
+        TargetStructType target;
+        ReturnErrorOnFailure(decoder.Decode(target));
+        if (target.nodeId.HasValue())
+        {
+            err = BindingManager::GetInstance().UnicastBindingCreated(target.fabricIdx, target.nodeId.Value());
+            if (err != CHIP_NO_ERROR)
+            {
+                ChipLogProgress(Zcl,
+                                "Binding: Failed to create session for unicast binding to device " ChipLogFormatX64
+                                ": %" CHIP_ERROR_FORMAT,
+                                ChipLogValueX64(target.nodeId.Value()), err.Format());
+            }
+        }
+        AddBindingEntry(target, path.mEndpointId);
     }
-
-    emberDeleteBinding(bindingIndex);
-    emberAfSendImmediateDefaultResponse(EMBER_ZCL_STATUS_SUCCESS);
-    return true;
+    return CHIP_ERROR_UNSUPPORTED_CHIP_FEATURE;
```

Comment:
We are reaching this for any attempt to write a nonempty list, because all such attempts go through the AppendItem case.  Simple reproduction:
```
chip-tool binding write binding-list '[{"fabricIdx": 0, "nodeId": 100, "endpointId": 5}]' node-id 1
```
Please actually test this code.  And add automated tests.